langlib/langlib-test/src/test/resources/test-src/regexp_test.bal

```diff
@@ -1311,6 +1325,313 @@ function testTranslatingDiffNodesInCharClass() {
+
+    string pattern17 = "(?im-s:abc";
+    string:RegExp|error reg17 = trap re `${pattern17}`;
+    assertTrue(reg17 is error);
+    assertEquality("{ballerina}RegularExpressionParsingError", (<error>reg17).message());
+    assertEquality("missing close parenthesis ')' token in insertion substring '(?im-s:abc'", 
+        <string>checkpanic (<error>reg17).detail()["message"]);
+    
+    string pattern18 = "\\p";
+    string:RegExp|error reg18 = trap re `${pattern18}`;
+    assertTrue(reg18 is error);
+    assertEquality("{ballerina}RegularExpressionParsingError", (<error>reg18).message());
+    assertEquality("missing open brace '{' token in insertion substring '\\p'", 
+        <string>checkpanic (<error>reg18).detail()["message"]);
+        
+    string pattern19 = "\\p{Ll";
+    string:RegExp|error reg19 = trap re `${pattern19}`;
+    assertTrue(reg19 is error);
+    assertEquality("{ballerina}RegularExpressionParsingError", (<error>reg19).message());
+    assertEquality("missing close brace '}' token in insertion substring '\\p{Ll'", 
+        <string>checkpanic (<error>reg19).detail()["message"]);
+        
+    string pattern20 = "abc\\w[a-!]";
+    string:RegExp|error reg20 = trap re `${pattern20}`;
+    assertTrue(reg20 is error);
+    assertEquality("{ballerina}RegularExpressionParsingError", (<error>reg20).message());
+    assertEquality("invalid character class range 'a'-'!' in insertion substring 'abc\\w[a-!]'", 
+        <string>checkpanic (<error>reg20).detail()["message"]);
+        
+    string pattern21 = "abc\\w[a--]";
+    string:RegExp|error reg21 = trap re `${pattern21}`;
+    assertTrue(reg21 is error);
+    assertEquality("{ballerina}RegularExpressionParsingError", (<error>reg21).message());
+    assertEquality("invalid character class range 'a'-'-' in insertion substring 'abc\\w[a--]'", 
+        <string>checkpanic (<error>reg21).detail()["message"]);
+        
+    string pattern22 = "abc[a-z[A-Z]]";
+    string:RegExp|error reg22 = trap re `${pattern22}`;
+    assertTrue(reg22 is error);
+    assertEquality("{ballerina}RegularExpressionParsingError", (<error>reg22).message());
+    assertEquality("missing backslash before ']' token in insertion substring 'abc[a-z[A-Z]]'", 
+        <string>checkpanic (<error>reg22).detail()["message"]);
+    
+    string pattern23 = "abc[a-zZ-A]";
+    string:RegExp|error reg23 = trap re `${pattern23}`;
+    assertTrue(reg23 is error);
+    assertEquality("{ballerina}RegularExpressionParsingError", (<error>reg23).message());
+    assertEquality("invalid character class range 'Z'-'A' in insertion substring 'abc[a-zZ-A]'", 
+        <string>checkpanic (<error>reg23).detail()["message"]);             
+}
+
+function testCharClassesWithMultipleRangesAndAtoms() returns error? {
+    string:RegExp pattern1 = check regexp:fromString("[a-zA-Z0-9]");
+    assertTrue(pattern1 is string:RegExp);
+    if (pattern1 is string:RegExp) {
```

Comment:
We can remove this check and use a cast, because we check the assertion in L1563